be/src/util/cpu_usage_info.cpp

```diff
@@ -0,0 +1,61 @@
+// This file is licensed under the Elastic License 2.0. Copyright 2021 StarRocks Limited.
+
+#include "util/cpu_usage_info.h"
+
+#include <glog/logging.h>
+
+#include <cstdio>
+#include <thread>
+
+#include "util/defer_op.h"
+#include "util/time.h"
+
+namespace starrocks {
+
+const int CpuUsageRecorder::NUM_HARDWARE_CORES = std::thread::hardware_concurrency();
+const int CpuUsageRecorder::SECOND_CLOCK_TICK = sysconf(_SC_CLK_TCK);
+
+CpuUsageRecorder::CpuUsageRecorder() : _timestamp{0, MonotonicNanos()}, _proc_time{0, _get_proc_time()} {}
+
+CpuUsageRecorder::~CpuUsageRecorder() {
+    if (_line_ptr != nullptr) {
+        free(_line_ptr);
+    }
+}
+
+void CpuUsageRecorder::update_interval() {
+    _curr_idx = (_curr_idx + 1) % 2;
+    _proc_time[_curr_idx] = _get_proc_time();
+    _timestamp[_curr_idx] = MonotonicNanos();
+}
+
+int CpuUsageRecorder::cpu_used_permille() const {
```

Comment:
what does permille mean? suggest to use more verbose name instead of abbr. to make the code more understandable.

Reply:
`permille` is not a abbr., it means `‰` or in thousand according to https://en.wikipedia.org/wiki/Per_mille.